scripts/doc-coverage.py

```
@@ -0,0 +1,407 @@
+    """
+    try:
+        per_file = json.loads(output)
+    except json.JSONDecodeError as error:
+        detail = (
+            f"cargo rustdoc for {target.package} {target.kind}"
+            f" ({target.name or 'lib'}) did not emit coverage JSON: {error}"
+        )
+        raise RuntimeError(detail) from error
+    try:
+        if not isinstance(per_file, dict):
+            raise TypeError("expected an object")
```

Ruff (0.16.2)

[warning] Abstract `raise` to an inner function

(TRY301)